### PR TITLE
Update on self training with a few ground_truth samples and multiple thresholds

### PR DESCRIPTION
**Pull request by BruceZZhao**
1. Created a new loader "gt_sample_loader" with 20 samples taken from the validation set as "manually" labeled ground truth for evaluation
2. Adding multiple thresholds for fair evaluation of CAMs
3. Fixed previous problems in calculating IOU scores
4. Adding back the "Classical" way of getting CAM from a classical CNN model with Global Average Pooling in CamManager
5. Adding a small eps in compute_iou_and_f1() to avoid getting nan F1 score (previously it will occasionally get nan when threshold is large, because precision and recall can both be 0.)

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -580,7 +663,7 @@ def fit_sgd_pixel(
         )
 
     @staticmethod
-    def predict_pixel_classification_dataset(
+    def predict_segmentation_dataset(
```
Comment (timfarkas) [inline models.py:666]: this method seems to be duplicate?

**Review by timfarkas — CHANGES_REQUESTED** [review]
looks good overall!
only thing: I don't quite understand the difference between the methods "predict_segmentation_dataset_on_current_loader" and "predict_segmentation_dataset". both take a loader as an argument. could you merge them/ remove one/ or change naming so its more clear which does what?

File: models.py (hunk 1 above)
Comment (BruceZZhao) [reply models.py]: The  predict_segmentation_dataset() simply use the predicted labels with filter to generate new data, mainly working with add-on method for getting new dataset. The predict_segmentation_dataset_on_current_loader() not only use predicted labels, but also use old labels to generate a new dataset, which does not need add-on as it already capture information from old dataset. As I might test both of them, I prefer to keep them for now. And for clarity, I have changed their names into predict_segmentation_dataset_with_basicfilter() and predict_segmentation_dataset_with_mixlabel(), respectively.
